src/viam/app/data/client.py

```diff
@@ -290,10 +328,226 @@ async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -
+        """
+        sensor_contents = [None] * len(tabular_data)
+        if timestamps:
+            assert len(timestamps) == len(tabular_data)
+
+        for i in range(len(tabular_data)):
+            s = Struct()
+            s.update(tabular_data[i])
+            sensor_contents[i] = SensorData(
+                metadata=SensorMetadata(
+                    time_requested=timestamps[i][0] if timestamps and timestamps[i][0] else None,
+                    time_received=timestamps[i][1] if timestamps and timestamps[i][1] else None
+                ),
+                struct=s
+            )
+
+        metadata = UploadMetadata(
+            part_id=part_id,
+            component_type=component_type,
+            component_name=component_name,
+            method_name=method_name,
+            type=DataType.DATA_TYPE_TABULAR_SENSOR,
+            file_name=None,  # Not used in app.
+            method_parameters=method_parameters if method_parameters else None,
+            file_extension=file_extension if file_extension else None,
+            tags=tags if tags else None,
+        )
+        _: DataCaptureUploadResponse = await self._data_capture_upload(metadata=metadata, sensor_contents=sensor_contents)
+
+    async def _data_capture_upload(self, metadata: UploadMetadata, sensor_contents: List[SensorData]) -> DataCaptureUploadResponse:
+        request = DataCaptureUploadRequest(metadata=metadata, sensor_contents=sensor_contents)
+        response: DataCaptureUploadResponse = await self._data_sync_client.DataCaptureUpload(request, metadata=self._metadata)
+        return response
+
+    async def file_upload(
+        self,
+        part_id: str,
+        component_type: Optional[str],
+        component_name: Optional[str],
+        method_name: Optional[str],
+        file_name: Optional[str],
+        method_parameters: Optional[Mapping[str, Any]],
+        file_extension: Optional[str],
+        tags: Optional[List[str]],
+        data: Optional[bytes]
+    ) -> None:
+        """Upload arbitrary file data.
+
+        Sync file data that may be stored on a robot along with the relevant metadata to app.viam.com.
+
+        Args:
+            part_id (str): Part ID of the component associated with the file.
+            component_type (Optional[str]): Optional type of the component associated with the file (i.e., "movement_sensor").
+            component_name (Optional[str]): Optional name of the component associated with the file.
+            method_name (Optional[str]): Optional name of the method associated with the file.
+            file_name (Optional[str]): Optional name of the file.
+            method_parameters (Optional[str]): Optional dicitonary of the method parameters. No longer in active use.
+            file_extension (Optional[str]): Optional file extension.
+            tags (Optional[List[str]]): Optional list of tags to allow for tag-based filtering.
```

Comment:
how does `tags` work here? since it's an upload and not fetching anything, it doesn't really make sense to me what `tag-based filtering` is.

Reply:
I think it attaches tags to the data. You're right my wording here is confusing. I mean that in the future when trying to retrieve this data, you can filter by the tags added. I will rephrase.